### PR TITLE
Chromosome data retrieval for all eukaryotes

### PR DESCRIPTION
**Pull request by ProjectProgramAMark**

### CODE REVIEW

**Comment by eweitz** [comment]
This pull request addresses the first task of #45 -- data retrieval for all eukaryotic chromosomes.

@ProjectProgramAMark, the diff on your commit is not visible on GitHub because it's too large.  https://github.com/eweitz/ideogram/pull/51/files currently says:

> 2,173 additions, 2,031 deletions not shown because the diff is too large. Please use a local Git client to view these changes.

Viewing your changes on the command line, I see the diff is mostly whitespace noise.  Even ignoring whitespace on the diff (`git diff -w HEAD^1`), though, I see style changes that are unrelated to this task, e.g.:

```
-    bands[i]["px"] = {"start": pxStop, "stop": pxStop + width, "width": width};
+        bands[i]["px"] = {
+            "start": pxStop,
+            "stop": pxStop + width,
+            "width": width
+        };
```

Please update your copy of `ideogram.js` to only add `getChromosomes` as an instance method on Ideogram, without adjusting the rest of the file's whitespace or object style.  This will make it possible to comment on the PR diff in GitHub, better preserve code history, etc.

**Comment by ProjectProgramAMark** [comment]
@eweitz ok this should be good. please take a look.

File: src/js/ideogram.js [hunk 1]
```diff
@@ -2129,6 +2129,90 @@ Ideogram.prototype.getTaxids = function() {
   return taxids;
 };
 
+/**
+ * Returns names and lengths of chromosomes fir an organism's best-known genome assembly
+ */
+Ideogram.prototype.getChromosomes = function(organismName) {
+    var output = [];
+    console.log("organismName is: ", organismName);
```
Comment (eweitz) [inline src/js/ideogram.js]: Please remove all instances of `console.log` in `getChromosomes`.

File: src/js/ideogram.js [hunk 2]
```diff
@@ -2129,6 +2129,90 @@ Ideogram.prototype.getTaxids = function() {
...
+    var organism = organismName.split(" ");
+    console.log("organism array is: ", organism);
+    // Querying NCBI database for genome
+    var link = "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=assembly&retmode=json&term=";
+    for (var i = 0; i < organism.length; i++) {
+        link += organism[i];
+        link += "%20";
+    }
+    link += "AND%20(%22latest%20refseq%22[filter])%20AND%20%22chromosome%20level%22[filter]";
+    d3.xhr(link, function(data) {
```
Comment (eweitz) [inline src/js/ideogram.js]: I originally recommended `d3.xhr`, but it occurs to me `d3.json` would be better for these requests.  That way, D3 will parse the response as JSON without us having to invoke `JSON.parse`.  Please replace `d3.xhr` calls in this method with `d3.xhr`, and remove those calls to `JSON.parse`.

File: src/js/ideogram.js [hunk 3]
```diff
@@ -2129,6 +2129,90 @@ Ideogram.prototype.getTaxids = function() {
...
+                        var chromosomeLength = (chromosomes[x].slen).toString();
+                        // console.log("type of length: ", typeof(chromosomeLength));
+                        // console.log("length: ", chromosomes[x].slen);
+                        var obj = {
+                            "name": chromosomeName,
+                            "length": chromosomeLength
+                        };
+                        output.push(obj);
+                    }
+                    for (var i = 0; i < output.length; i++) {
```
Comment (eweitz) [inline src/js/ideogram.js]: `return output;` once the full array of chromosome lengths and names is ready

**Comment by ProjectProgramAMark** [comment]
@eweitz done

**Comment by eweitz** [comment]
:tada:

Thank you, @ProjectProgramAMark!  I'll put together an [AUTHORS](https://github.com/jquery/jquery/blob/master/AUTHORS.txt) file.  What name and email can I use to acknowledge you?

**Comment by ProjectProgramAMark** [comment]
@eweitz 
Name: Mark Moussa
Email: projectprogramamark@gmail.com

**Pull request merged**